blocks/library/image/style.scss

```diff
@@ -9,11 +9,19 @@
 	&[data-align="left"] {
 		float: left;
 		margin-right: $block-padding;
+
+		@include break-medium() {
+			margin-left: calc( 50% - #{ $visual-editor-max-width / 2 } );
+		}
```

Comment:
Good aside. And no, I've been thinking of it this way:

- Full-width can be implemented in many different ways, we've chosen one of many ways
- Theme authors should probably opt-in to the feature, probably something like `add_theme_support( 'full-width-images' );`
- When the theme has opted in, images simply get `align-full-width` as a CSS class. Or something in that vein.